x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_creation/common_components.cy.ts

```diff
@@ -0,0 +1,100 @@
+
+import { ruleFields } from '../../../data/detection_engine';
+import { getTimeline } from '../../../objects/timeline';
+
+import {
+  ABOUT_CONTINUE_BTN,
+  ABOUT_EDIT_BUTTON,
+  CUSTOM_QUERY_INPUT,
+  DEFINE_CONTINUE_BUTTON,
+  DEFINE_EDIT_BUTTON,
+  RULE_NAME_INPUT,
+  SCHEDULE_CONTINUE_BUTTON,
+} from '../../../screens/create_new_rule';
+import { RULE_NAME_HEADER } from '../../../screens/rule_details';
+import { createTimeline } from '../../../tasks/api_calls/timelines';
+import { deleteAlertsAndRules } from '../../../tasks/common';
+import {
+  createAndEnableRuleOnly,
+  expandAdvancedSettings,
+  fillCustomInvestigationFields,
+  fillDescription,
+  fillFalsePositiveExamples,
+  fillFrom,
+  fillNote,
+  fillReferenceUrls,
+  fillRiskScore,
+  fillRuleName,
+  fillRuleTags,
+  fillSeverity,
+  fillThreat,
+  fillThreatSubtechnique,
+  fillThreatTechnique,
+  importSavedQuery,
+} from '../../../tasks/create_new_rule';
+import { login, visit } from '../../../tasks/login';
+import { RULE_CREATION } from '../../../urls/navigation';
+
+// This test is meant to test touching all the common various components in rule creation
+// to ensure we don't miss any changes that maybe affect one of these more obscure UI components
+// in the creation form. For any rule type specific functionalities, please include
+// them in the relevant /rule_creation/[RULE_TYPE].cy.ts test.
+describe('Common rule creation components', { tags: ['@ess', '@serverless'] }, () => {
+  beforeEach(() => {
+    deleteAlertsAndRules();
+    createTimeline(getTimeline())
+      .then((response) => {
+        return response.body.data.persistTimeline.timeline.savedObjectId;
+      })
+      .as('timelineId');
+    login();
+  });
+
+  it('Creates and enables a rule', function () {
+    visit(RULE_CREATION);
+
+    cy.log('Filling define section');
+    importSavedQuery(this.timelineId);
+    cy.get(DEFINE_CONTINUE_BUTTON).click();
+
+    cy.log('Filling about section');
+    fillRuleName();
+    fillDescription();
+    fillSeverity();
+    fillRiskScore();
+    fillRuleTags();
+    expandAdvancedSettings();
+    fillReferenceUrls();
+    fillFalsePositiveExamples();
+    fillThreat();
+    fillThreatTechnique();
+    fillThreatSubtechnique();
+    fillCustomInvestigationFields();
+    fillNote();
+    cy.get(ABOUT_CONTINUE_BTN).click();
+
+    cy.log('Filling schedule section');
+    fillFrom();
+
+    // expect define step to repopulate
+    cy.get(DEFINE_EDIT_BUTTON).click();
+    cy.get(CUSTOM_QUERY_INPUT).should('have.value', ruleFields.ruleQuery);
+    cy.get(DEFINE_CONTINUE_BUTTON).should('exist').click();
+
+    // expect about step to populate
+    cy.get(ABOUT_EDIT_BUTTON).click();
+    cy.get(RULE_NAME_INPUT).invoke('val').should('eql', ruleFields.ruleName);
+    cy.get(ABOUT_CONTINUE_BTN).should('exist').click();
+    cy.get(SCHEDULE_CONTINUE_BUTTON).click();
+
+    createAndEnableRuleOnly();
+
+    cy.get(RULE_NAME_HEADER).should('contain', ruleFields.ruleName);
```

Comment:
now, that we don't check if rule details displayed, after rule was created, should we introduce one test case on common scenarios for this?
So, we can be sure, that after user creates rule, they get redirected to details page and nothing is broken in this workflow

Reply:
If it's ok, I was able to clean up and add some scenarios to test details, but it's added a lot more changes. If it's ok, I'd prefer to put this PR in as is as it's confirmed to be passing consistently and then follow up with more changes/test cases.